pkg/apis/apps/types.go

```diff
@@ -198,6 +198,9 @@ type StatefulSetStatus struct {
 
 	// Represents the latest available observations of a statefulset's current state.
 	Conditions []StatefulSetCondition
+
+	// volumeClaimTemplateMismatches is the number of persistent volume claims in use which do not match their template
+	VolumeClaimTemplateMismatches int32
```

Comment:
Why not flag this situation as one of the StatefulSetConditions ?

Reply:
that way we would get more details of not only the mismatch, but what type of mismatch

Reply:
@krmayankk Hm, well, mostly because I apparently didn't understand what a Condition is :) I interpreted it as something to do with conditionals, but from a quick search that is not the case. Is there some guidelines around what should be used when? And some closer description of the Condition concept?

Reply:
Agree with krmayankk. This is a persistent condition. I wouldn't add this as a status field. If we do implement resizable PVCs/PVs then field has no meaning, but we are stuck with it. A condition can be deprecated and removed in the future without breaking backward compatibility.

Reply:
Understood. Will rework the PR!

Reply:
+1 for using condition